sycl/test/aot/spec_const_aot.cpp

```diff
@@ -38,7 +38,6 @@ int main(int argc, char **argv) {
   cl::sycl::experimental::spec_constant<int32_t, MyInt32Const> i32 =
       prog.set_spec_constant<MyInt32Const>(10);
 
-  prog.build_with_kernel_type<Kernel>();
```

Comment:
Now, we do not support use  “explicit JIT API” in AOT mode. Since it is an incorrect attempt to compile an already compiled kernel.
It is necessary because with new changes it will be thrown an exception, so, each test that uses in AOT mode, for example: build_with_kernel_type(), compile_with_kernel_type() will fail.

Reply:
> Now, we do not support use “explicit JIT API” in AOT mode. Since it is an incorrect attempt to compile an already compiled kernel.

this is not obvious to me. I can see your point, but  with this approach user has to worry about separation of sources compiled in JIT and AOT mode. Spec constants themselves are not supported in AOT mode and are replaced by defaults instead of generating an error for code maintainability. @Ruyk, would appreciate your comments here.

Reply:
A user needs to worry about JIT and AOT mode anyway, at least due to different values of spec constants. We need some solution which is aligned with the SYCL spec and CTS and works with h spec constants extension. As a short term solution/ workaournd for the program class we could unofficially allow calling build_with_kernel_type without arguments even if SPIRV is not available?

Reply:
> A user needs to worry about JIT and AOT mode anyway, at least due to different values of spec constants.

Under worry I meant actually creating separate code versions to avoid crash.

> As a short term solution/ workaournd for the program class we could unofficially allow calling build_with_kernel_type without arguments even if SPIRV is not available?

makes sense to me.